homeassistant/components/vasttrafik/sensor.py

```diff
@@ -60,49 +67,150 @@
+                "integration_title": "Västtrafik",
+            },
+        )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: VasttrafikConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up Västtrafik sensor based on a config entry."""
+    planner = entry.runtime_data
+
+    for subentry in entry.subentries.values():
+        if subentry.subentry_type == "departure_board":
+            async_add_entities(
+                [
+                    VasttrafikDepartureSensor(
+                        planner,
+                        subentry.data[CONF_NAME],
+                        subentry.data[CONF_FROM],
+                        subentry.data.get(CONF_HEADING, ""),
+                        subentry.data[CONF_LINES],
+                        subentry.data[CONF_TRACKS],
+                        subentry.data[CONF_DELAY],
+                        entry.entry_id,
+                        subentry.subentry_id,
+                        subentry.subentry_id,
+                    )
+                ],
+                update_before_add=True,
+                config_subentry_id=subentry.subentry_id,
+            )
+
 
 class VasttrafikDepartureSensor(SensorEntity):
     """Implementation of a Vasttrafik Departure Sensor."""
 
     _attr_attribution = "Data provided by Västtrafik"
     _attr_icon = "mdi:train"
+    _attr_has_entity_name = True
 
-    def __init__(self, planner, name, departure, heading, lines, delay):
+    def __init__(
+        self,
+        planner,
+        name,
+        departure,
+        heading,
+        lines,
+        tracks,
+        delay,
+        config_entry_id,
+        unique_suffix=None,
+        subentry_id=None,
+    ) -> None:
         """Initialize the sensor."""
         self._planner = planner
-        self._name = name or departure
-        self._departure = self.get_station_id(departure)
-        self._heading = self.get_station_id(heading) if heading else None
+        if name:
+            self._attr_name = name
+        else:
+            self._attr_translation_key = "departures"
+            self._attr_translation_placeholders = {"station": departure}
+        self._departure_name = departure
+        self._heading_name = heading
+        self._departure: dict[str, str] | None = (
+            None  # Will be resolved on first update
+        )
+        self._heading: dict[str, str] | None = None  # Will be resolved on first update
         self._lines = lines if lines else None
+        self._tracks = tracks if tracks else None
         self._delay = timedelta(minutes=delay)
-        self._departureboard = None
-        self._state = None
-        self._attributes = None
+        self._departureboard: Any = None
+        self._state: str | None = None
+        self._attributes: dict[str, Any] | None = None
+
+        # Create device info for departure board subentries (similar to Ollama pattern)
+        if subentry_id:
```

Comment:
This is unneccessary as you have a subentry_id (which I suggest to remove as you don't need the check at all).